folio-spring-base/src/test/java/org/folio/spring/DefaultFolioExecutionContextTest.java

```diff
@@ -79,4 +79,20 @@ void testExtractionFromHeadersWhenTokenHeaderNotExists() {
     assertThat(actual).satisfies(contextRequirements);
   }
 
+  @Test
+  void testRemovalNonOkapiAndCaseInsensitiveDuplicatedHeaders() {
```

Comment:
The test doesn't check for removing duplicates. The test checks if `okapiHeaders` contains only headers that start with 'x-okapi`.